### PR TITLE
FE: Unit tests for monitoringPoliciesStore

### PR DESCRIPTION
**Pull request by faizanjanjua**
## Description
Add tests/store/Views/monitoringPoliciesStore.test.ts and add tests for all methods.

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2413

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: ui/tests/store/Views/monitoringPolicyStore.test.ts [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+  afterEach(() => {
+    vi.restoreAllMocks()
+  })
+
+
+  it('monitoringPolicies.displayPolicyForm complete successfully', async () => {
+    const { useMonitoringPoliciesStore } = await import('@/store/Views/monitoringPoliciesStore')
+    const monitoringPolicies = useMonitoringPoliciesStore()
+    const mockData: Policy = {isDefault: true, ...mockMonitoringPolicy}
+    vi.spyOn(monitoringPolicies, 'displayPolicyForm')
```
Comment (synqotik) [inline ui/tests/store/Views/monitoringPolicyStore.test.ts:112]: Remove all these `vi.spyOn` if you can. When using `createTestingPinia`, all items in the `store` have spies automatically attached. Also in some cases, the `spyOn` doesn't match the method you are checking, so it's confusing.

File: ui/tests/store/Views/monitoringPolicyStore.test.ts [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+      notifyByWebhooks: true,
+      rules: [],
+      tags: ['default']
+    },
+    listMonitoryPolicies: []
+  }
+})
+
+describe('Node Status Store', () => {
+  let stubActionsFlag = false // Flag to control stubActions
```
Comment (synqotik) [inline ui/tests/store/Views/monitoringPolicyStore.test.ts:97]: Let's try this. Make 2 `describe` blocks under the main one - one where `stubActions` is `false`, the other when `stubActions` is `true`. You can call one `describe('when stubActions is false')` and `describe('when stubActions is true')`.

Then in `beforeEach`, just have `createTestingPinia({ stubActions: false })` (or `true`) and don't need `stubActionsFlag`.

Also remove any other `beforeAll(() => createTestingPinia())` unless it's absolutely needed.

File: ui/tests/store/Views/monitoringPolicyStore.test.ts [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+    store.selectedPolicy = {...mockMonitoringPolicy, isDefault: false}
+    store.selectedRule = mockMonitoringPolicy && mockMonitoringPolicy.rules ? mockMonitoringPolicy?.rules[0] : {}
+
+    vi.spyOn(store, 'addNewCondition')
+    await store.addNewCondition()
+    expect(store.addNewCondition).toHaveBeenCalledOnce()
+  })
+
+
+  it('monitoringPolicies.getMonitoringPolicies complete successfully', async () => {
```
Comment (synqotik) [inline ui/tests/store/Views/monitoringPolicyStore.test.ts:310]: See above comment, put these 2 items that have `stubActions: true` into a separate nested `describe`.

File: ui/tests/store/Views/monitoringPolicyStore.test.ts [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+    const { useMonitoringPoliciesStore } = await import('@/store/Views/monitoringPoliciesStore')
+    const store = useMonitoringPoliciesStore()
+    store.selectedPolicy = {...mockMonitoringPolicy, isDefault: false}
+    store.selectedRule = mockMonitoringPolicy && mockMonitoringPolicy.rules ? mockMonitoringPolicy?.rules[0] : {}
+    vi.spyOn(store, 'updateCondition')
+    store.deleteCondition('2')
+    expect(store.deleteCondition).toHaveBeenCalledWith('2')
+  })
+
+  it('monitoringPolicies.deleteCondition complete successfully', async () => {
```
Comment (synqotik) [inline ui/tests/store/Views/monitoringPolicyStore.test.ts:253]: Should be called `test('monitoringPolicies.validateRule completes successfully'`

File: ui/tests/store/Views/monitoringPolicyStore.test.ts [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+    const store = useMonitoringPoliciesStore()
+    store.selectedPolicy = {...mockMonitoringPolicy, isDefault: false}
+    store.selectedRule = mockMonitoringPolicy && mockMonitoringPolicy.rules ? mockMonitoringPolicy?.rules[0] : {}
+
+    vi.spyOn(store, 'resetDefaultConditions')
+    await store.resetDefaultConditions()
+    expect(store.resetDefaultConditions).toHaveBeenCalledOnce()
+  })
+
+  it('monitoringPolicies.addNewCondition complete successfully', async () => {
```
Comment (synqotik) [inline ui/tests/store/Views/monitoringPolicyStore.test.ts:297]: This looks like a duplicate, already implemented above, so you can delete it here.

**Review by synqotik — CHANGES_REQUESTED** [review]
Added some comments.

In many cases, we can test the actual functionality. But let's get this approved and merged, then we can open another issue to make the tests more robust.

File: ui/tests/store/Views/monitoringPolicyStore.test.ts (hunk 2 above)
Comment (faizanjanjua) [reply ui/tests/store/Views/monitoringPolicyStore.test.ts:97]: done

File: ui/tests/store/Views/monitoringPolicyStore.test.ts (hunk 3 above)
Comment (faizanjanjua) [reply ui/tests/store/Views/monitoringPolicyStore.test.ts:310]: done

File: ui/tests/store/Views/monitoringPolicyStore.test.ts (hunk 4 above)
Comment (faizanjanjua) [reply ui/tests/store/Views/monitoringPolicyStore.test.ts:253]: done

File: ui/tests/store/Views/monitoringPolicyStore.test.ts (hunk 5 above)
Comment (faizanjanjua) [reply ui/tests/store/Views/monitoringPolicyStore.test.ts:297]: done

**Review by synqotik — APPROVED** [review]

**Pull request merged**